Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/audit/transform/DocRetrieveAuditTransforms.java

```diff
@@ -0,0 +1,300 @@
+            && request.getDocumentRequest().get(0).getDocumentUniqueId() != null) {
+            documentUniqueId = request.getDocumentRequest().get(0).getDocumentUniqueId();
+        } else {
+            LOG.error("DocumentId doesn't exist in the received RetrieveDocumentSetRequestType message");
+        }
+        return documentUniqueId;
+    }
+
+    private String getRepositoryUniqueIdFromRequest(RetrieveDocumentSetRequestType request) {
+        String repositoryUniqueId = null;
+
+        if (request != null
+            && request.getDocumentRequest() != null
+            && request.getDocumentRequest().get(0) != null
+            && request.getDocumentRequest().get(0).getRepositoryUniqueId() != null) {
+            repositoryUniqueId = request.getDocumentRequest().get(0).getRepositoryUniqueId();
+        } else {
+            LOG.error("DocumentId doesn't exist in the received RetrieveDocumentSetRequestType message");
+        }
+        return repositoryUniqueId;
+    }
+
+    private String getHomeCommunityIdFromRequest(RetrieveDocumentSetRequestType request) {
+        String homeCommunityId = null;
+
+        if (request != null
+            && request.getDocumentRequest() != null
+            && request.getDocumentRequest().get(0) != null
+            && request.getDocumentRequest().get(0).getHomeCommunityId() != null) {
+            homeCommunityId = request.getDocumentRequest().get(0).getHomeCommunityId();
+        } else {
+            LOG.error("HomeCommunityId doesn't exist in the received RetrieveDocumentSetRequestType message");
+        }
+        return homeCommunityId;
+    }
+
+    /**
+     * This method gets unique repository unique id from response
+     *
+     * @param response
+     * @return
+     */
+    private String getRepositoryUniqueIdFromResponse(RetrieveDocumentSetResponseType response) {
+        String repositoryUniqueId = null;
+
+        if (response != null
+            && response.getDocumentResponse() != null
+            && response.getDocumentResponse().get(0) != null
+            && response.getDocumentResponse().get(0).getRepositoryUniqueId() != null) {
+            repositoryUniqueId = response.getDocumentResponse().get(0).getRepositoryUniqueId();
+        } else {
+            LOG.error("DocumentId doesn't exist in RetrieveDocumentSetResponseType message");
+        }
+        return repositoryUniqueId;
+    }
+
+    /**
+     * This method gets home community id from response
+     *
+     * @param response
+     * @return
+     */
+    private String getHomeCommunityIdFromResponse(RetrieveDocumentSetResponseType response) {
+        String homeCommunityId = null;
+
+        if (response != null
+            && response.getDocumentResponse() != null
+            && response.getDocumentResponse().get(0) != null
+            && response.getDocumentResponse().get(0).getHomeCommunityId() != null) {
+            homeCommunityId = response.getDocumentResponse().get(0).getHomeCommunityId();
```

Comment:
Also favor an extra line break between multi-line conditionals and the code block it surrounds -- but again, don't worry about fixing for this PR, just keep it in mind for the future.